core/services/vrf/v2/listener_v2_log_processor.go

```diff
@@ -1162,6 +1162,25 @@ func (lsn *listenerV2) simulateFulfillment(
 					res.proof = FromV2Proof(m["proof"].(vrf_coordinator_v2.VRFProof))
```

Comment:
Would it be possible to reuse this if block by making the condition:

```
if trr.Task.Type() == pipeline.TaskTypeVRFV2 || trr.Task.Type() == pipeline.TaskTypeVRFV2Plus {

and having an if condition around proof

if trr.Task.Type() == pipeline.TaskTypeVRFV2 {
   res.proof = FromV2Proof(m["proof"].(vrf_coordinator_v2.VRFProof))
} else {
  res.proof = FromV2PlusProof(m["proof"].(vrf_coordinator_v2plus_interface.IVRFCoordinatorV2PlusInternalProof))
}
```

I don't see any other difference between the two code blocks and we could re-use much of the previous logic without code duplication.

Reply:
It's possible. But I think it's easier to read and manage if they were 2 separate blocks than if else, shared code, if else, maybe more shared code. The 2 versions while doing mostly the same thing might not necessarily have logic that is always applicable to both versions. If the bulk of the logic in the future is shared, then yes, it's better to refactor to what you suggested.